File: problems/849/solution.go
package main
func maxDistToClosest(seats []int) int {
    ans := 0
    
    for seats[ans] != 1 {
        ans++
    }
    
    for i := ans; i < len(seats); i++ {
        if seats[i] == 0 {
            continue
        }
        
        j := i
        
        for i + 1 < len(seats) && seats[i + 1] != 1 {
            i++
        }
        
        if i == len(seats) - 1 {
            ans = max(ans, i - j)
        } else {
            ans = max(ans, (i - j + 1) / 2)
        }
    }
    
    return ans
}

func max(a, b int) int {
    if a > b {
        return a
    }
    
    return b
}

File: problems/849/solution.cpp
class Solution {
public:
    int maxDistToClosest(vector<int>& seats) {
        int maxDist = 0;
        
        while (seats[maxDist] == 0) {
            maxDist++;
        }
        
        for (int i = maxDist; i < seats.size(); i++) {
            if (!seats[i]) {
                continue;
            }
            
            int j = i;
                
            while (i + 1 < seats.size() && seats[i + 1] != 1) {
                i++;
            }

            if (i == seats.size() - 1 && i - j > maxDist) {
                maxDist = i - j;
            } else if ((i - j + 1) / 2 > maxDist) {
                maxDist = (i - j + 1) / 2;
            }
        }
        
        return maxDist;
    }
};